### DISCUSSION TITLE
Is exponentiation defined for integral types?

### DISCUSSION THREAD

**Discussion by ornamentist** [coalton-lang/coalton#645] [Help!] [answered]
I'd like to be able to raise `Integer` values to a positive `Integer` power. 

I think there's an `Exponentiable` typeclass which is not suitable for integral types? Ther seems to be current discussion around whether `expt`, `pow` or `**` are suitable function names for exponentiation?

Will any of these deliberations eventually apply to integral types?

**Comment by stylewarning** [comment, marked as answer]
How about [`^`](https://github.com/coalton-lang/coalton/blob/main/library/math/integral.lisp#L65), or `^^` if you want to raise to negative powers?

Admittedly the current state of affairs might seem a little confusing. I started writing a guide to the math functions and classes of Coalton but haven't finished it.

**Reply by ornamentist** [reply]
I'm happy to use whatever the Coalton developers recommend for this.  I'll try `^` and `^^`.

**Comment by ornamentist** [comment]
FWIW: When working with Haskell-derived languages and creating mathematical structures, I found this guide useful: https://a-guide-to-the-purescript-numeric-hierarchy.readthedocs.io/en/latest/introduction.html
